comms/src/blazingdb/manager/Context.cc

```diff
@@ -56,14 +64,21 @@ std::string Context::getLogicalPlan() const { return logicalPlan_; }
 
 uint32_t Context::getContextToken() const { return token_; }
 
-uint32_t Context::getContextCommunicationToken() const { return query_substep; }
+std::string Context::getContextCommunicationToken() const {
+  return std::to_string(kernel_id_) + "_"  + std::to_string(query_substep) ;   
+}
 
 void Context::incrementQueryStep() {
+  std::unique_lock<std::mutex> lock(increment_step_mutex);
```

Comment:
how can two processes be in contention for incrementing the queryStep of a context? If that happens it feels like thats a bug right? Isn't this supposed to be linear?, if its not linear for some reason what does that mean?

Reply:
is the query subset not set back to 0 whe you move onto the next step?

Reply:
so we even need query steps?